server/portal/apps/workspace/api/views.py

```diff
@@ -297,6 +297,11 @@ def post(self, request, *args, **kwargs):
                     homeDir = settings.PORTAL_DATAFILES_DEFAULT_STORAGE_SYSTEM['homeDir'].format(tasdir=tasdir, username=username)
                     job_post['archiveSystemDir'] = f'{homeDir}/tapis-jobs-archive/${{JobCreateDate}}/${{JobName}}-${{JobUUID}}'
 
+            execSystemId = job_post.get("execSystemId")
+            if not execSystemId:
```

Comment:
The only change we might want to consider is handling a `''` or `undefined` value for `job_post['appVersion']`. In these cases, we can also enter this `if` statement and get the `appVersion` from `_get_app()`